membership-attribute-service/app/services/AttributesFromZuora.scala

```diff
@@ -52,13 +55,24 @@ class AttributesFromZuora(implicit val executionContext: ExecutionContext, syste
           case Failure(e) => SafeLogger.error(scrub"Failed to retrieve attributes from cache for $identityId because $e")
         }
 
+    def userHasDigiSub(accountsWithSubscriptions: List[AccountWithSubscriptions]) =
+      accountsWithSubscriptions.flatMap(_.subscriptions).exists(_.asDigipack.isDefined)
+
     lazy val getAttrFromZuora =
       for {
         accounts <- EitherT(identityIdToAccounts(identityId))
-        subscriptions <- EitherT(if (accounts.size > 0) getSubscriptions(accounts, subscriptionsForAccountId) else Future.successful(\/.right[String, List[AccountWithSubscriptions]](Nil)))
+        subscriptions <- EitherT(
+          if (accounts.size > 0) getSubscriptions(accounts, subscriptionsForAccountId)
+          else Future.successful(\/.right[String, List[AccountWithSubscriptions]](Nil))
+        )
+        giftSubscriptions <- EitherT(
+          if(userHasDigiSub(subscriptions)) Future.successful(\/.right[String, List[GiftSubscriptionsFromIdentityIdRecord]](Nil))
+          else giftSubscriptionsForIdentityId(identityId)
+        )
+        giftAttributes = giftSubscriptions.headOption.map(record => ZuoraAttributes(identityId, DigitalSubscriptionExpiryDate = Some(record.TermEndDate)))
         maybeZAttributes <- EitherT(AttributesMaker.zuoraAttributes(identityId, subscriptions, paymentMethodForPaymentMethodId, forDate).map(\/.right[String, Option[ZuoraAttributes]]))
       } yield {
-        val maybeAttributes = maybeZAttributes.map(ZuoraAttributes.asAttributes(_))
+        val maybeAttributes = maybeZAttributes.orElse(giftAttributes).map(ZuoraAttributes.asAttributes(_))
```

Comment:
if they have another attributable product `maybeZAttributes` will come back as Some, so we wouldn't use the gift

Reply:
aattributable products are a few https://github.com/guardian/members-data-api/blob/e48de2a0a3b4388ed86e5904f2e5a513de68c0eb/membership-attribute-service/app/services/AttributesMaker.scala#L98-L102

Reply:
Thanks for spotting this